org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterEurotronicSpzb0001WindowOpen.java

```diff
@@ -0,0 +1,195 @@
+import org.openhab.binding.zigbee.handler.ZigBeeThingHandler;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.CommandResult;
+import com.zsmartsystems.zigbee.ZigBeeEndpoint;
+import com.zsmartsystems.zigbee.zcl.ZclAttribute;
+import com.zsmartsystems.zigbee.zcl.ZclAttributeListener;
+import com.zsmartsystems.zigbee.zcl.clusters.ZclThermostatCluster;
+import com.zsmartsystems.zigbee.zcl.protocol.ZclClusterType;
+import com.zsmartsystems.zigbee.zcl.protocol.ZclDataType;
+
+/**
+ * Converter for the thermostat system mode channel. The SystemMode attribute specifies the current operating mode of
+ * the thermostat,
+ *
+ * @author Robert Schmid - Initial Contribution
+ *
+ */
+public class ZigBeeConverterEurotronicSpzb0001WindowOpen extends ZigBeeBaseChannelConverter
+        implements ZclAttributeListener {
+    private Logger logger = LoggerFactory.getLogger(ZigBeeConverterEurotronicSpzb0001WindowOpen.class);
+
+    private final static int MANUFACTURER_EUROTRONIC = 0x1037;
+    private final static int ATTR_HOST_FLAGS = 0x4008;
+    private final static int WINDOW_OPEN_DISABLED = 16;
+    private final static int WINDOW_OPEN_ENABLED = 32;
+
+    private ZclThermostatCluster cluster;
+    private ZclAttribute attribute;
+
+    @Override
+    public Set<Integer> getImplementedClientClusters() {
+        return Collections.singleton(ZclThermostatCluster.CLUSTER_ID);
+    }
+
+    @Override
+    public Set<Integer> getImplementedServerClusters() {
+        return Collections.emptySet();
+    }
+
+    private ZclThermostatCluster getCluster(ZigBeeEndpoint endpoint) {
+      ZclThermostatCluster cluster = (ZclThermostatCluster) endpoint.getInputCluster(ZclThermostatCluster.CLUSTER_ID);
+      if (cluster == null) {
+        return null;
+      }
+
+      ZclAttribute hostFlagsAttribute = new ZclAttribute(cluster, ATTR_HOST_FLAGS, "Host Flags",
+        ZclDataType.UNSIGNED_24_BIT_INTEGER, false, true, true, true, MANUFACTURER_EUROTRONIC);
+      cluster.addAttributes(new HashSet<>(Arrays.asList(hostFlagsAttribute)));
+
+      return cluster;
+    }
+
+    @Override
+    public boolean initializeDevice() {
+        ZclThermostatCluster serverCluster = getCluster(endpoint);
+        if (serverCluster == null) {
+          logger.error("{}: Error opening device thermostat cluster", endpoint.getIeeeAddress());
+          return false;
+        }
+
+        try {
+            CommandResult bindResponse = bind(serverCluster).get();
+            if (bindResponse.isSuccess()) {
+                // Configure reporting
+                ZclAttribute attribute = serverCluster.getAttribute(ATTR_HOST_FLAGS);
+                CommandResult reportingResponse = attribute
```

Comment:
There still is one issue I wasn't yet able to resolve: Sometimes the following exception occurs when enabling the handler:

```
java.lang.IllegalArgumentException: You cannot append null data to a stream
	at com.zsmartsystems.zigbee.serialization.DefaultSerializer.appendZigBeeType(DefaultSerializer.java:43) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.field.AttributeReportingConfigurationRecord.serialize(AttributeReportingConfigurationRecord.java:361) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclFieldSerializer.serialize(ZclFieldSerializer.java:38) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.clusters.general.ConfigureReportingCommand.serialize(ConfigureReportingCommand.java:107) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.ZigBeeNetworkManager.sendCommand(ZigBeeNetworkManager.java:903) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.transaction.ZigBeeTransactionManager.send(ZigBeeTransactionManager.java:444) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.transaction.ZigBeeTransactionManager.sendNextTransaction(ZigBeeTransactionManager.java:682) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.transaction.ZigBeeTransactionManager.queueTransaction(ZigBeeTransactionManager.java:373) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.transaction.ZigBeeTransactionManager.sendTransaction(ZigBeeTransactionManager.java:352) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.ZigBeeNetworkManager.sendTransaction(ZigBeeNetworkManager.java:2030) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.ZigBeeNode.sendTransaction(ZigBeeNode.java:963) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.ZigBeeEndpoint.sendTransaction(ZigBeeEndpoint.java:596) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclCluster.sendCommand(ZclCluster.java:338) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclCluster.sendCommand(ZclCluster.java:349) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclCluster.setReporting(ZclCluster.java:1867) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclCluster.setReporting(ZclCluster.java:662) ~[bundleFile:?]
	at com.zsmartsystems.zigbee.zcl.ZclAttribute.setReporting(ZclAttribute.java:477) ~[bundleFile:?]
	at org.openhab.binding.zigbee.internal.converter.ZigBeeConverterEurotronicSpzb0001WindowOpen.initializeDevice(ZigBeeConverterEurotronicSpzb0001WindowOpen.java:96) ~[bundleFile:?]
	at org.openhab.binding.zigbee.handler.ZigBeeThingHandler.initializeDevice(ZigBeeThingHandler.java:514) ~[bundleFile:?]
	at org.openhab.binding.zigbee.handler.ZigBeeThingHandler.doNodeInitialisation(ZigBeeThingHandler.java:377) [bundleFile:?]
	at org.openhab.binding.zigbee.handler.ZigBeeThingHandler$1.call(ZigBeeThingHandler.java:227) [bundleFile:?]
	at org.openhab.binding.zigbee.handler.ZigBeeThingHandler$1.call(ZigBeeThingHandler.java:1) [bundleFile:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
19:44:59.119 [INFO ] [hab.event.ThingStatusInfoChangedEvent]
```

Reply:
Possibly this is considered an analog attribute, in which case there is another parameter that needs to go in the reporting configuration. If the attribute is analog, and you use the digital version of this method, then I think the library will try and send the extra parameter and it will be null.